nuget/lib/dependabot/nuget/update_checker/repository_finder.rb

```diff
@@ -163,15 +163,15 @@ def repos_from_config_file(config_file)
               }
             end
 
+          unless doc.css("configuration > packageSources > clear").any?
+            sources << { url: DEFAULT_REPOSITORY_URL, key: nil }
```

Comment:
Since the key is `nil` here I'm wondering how it could possible be removed by the disabled sources which rejects by key. Is there something else that would filter this out? Are we de-duping the the url somewhere?

Reply:
Good catch, it looks like `disabled_sources` is returning nil because of the `value="false"` in the test which happens to match the default registry key here and removes it.

Reply:
Good point, I looked into it. The function `disabled_sources` returns an array of the same number of elements it is provided as input because it is using map method inside `(doc.css("configuration > disabledPackageSources > add").select do |node|)` . So based on the number of entires in the`<disabledPackageSources>` section of nuget.config file it will return same number of elements .

So for the fixture which I have written it had two entries and the output of `disabled_sources` function is `[nil, 'nuget.org']`.

```
    <disabledPackageSources>
        <add key="MyRepo - ES" value="false" />
        <add key="nuget.org" value="true" />
    </disabledPackageSources>
```

Since the output of the `disabled_sources is [nil, 'nuget.org']` the below code is able to reject the key with nil value.
```
  sources.reject! do |s|
    disabled_sources.include?(s[:key])
  end
```

I can see where this will fail, when dependabot has access to public internet, no `<clear />` tag is provided in nuget.config file and the `<disabledPackageSources>` contains more then one entry with value as 'false'. In this case `disabled_sources` function will return `[nil]` and will remove the default `nuget.org` because we are setting the sources with key as nil `sources << { url: DEFAULT_REPOSITORY_URL, key: nil }`. A simple fix will be to add a key 'nuget.org' instead of `nil` to code `sources << { url: DEFAULT_REPOSITORY_URL, key: nil }`

```
    <disabledPackageSources>
        <add key="MyRepo - ES" value="false" />
    </disabledPackageSources>
```

Reply:
Push the changes.

Reply:
A simple fix is that a key 'nuget.org' instead of `nil` is added the to code `sources << { url: DEFAULT_REPOSITORY_URL, key: nil }`